imap_processing/lo/l1c/lo_l1c.py

```diff
@@ -31,34 +32,96 @@ def lo_l1c(dependencies: dict) -> list[Path]:
-            data_field_tup("OXYGEN_RATES"),
-            data_field_tup("EXPOSURE_TIME"),
-        ]
+        l1b_de = sci_dependencies["imap_lo_l1b_de"]
 
-    dataset: list[Path] = create_datasets(attr_mgr, logical_source, data_fields)  # type: ignore[arg-type]
-    # TODO Remove once data_fields input is removed from create_datasets
-    return dataset
+        l1b_goodtimes_only = filter_goodtimes(l1b_de, anc_dependencies)
+        pset = initialize_pset(l1b_goodtimes_only, attr_mgr, logical_source)
+    return [pset]
+
+
+def initialize_pset(
+    l1b_de: xr.Dataset, attr_mgr: ImapCdfAttributes, logical_source: str
+) -> xr.Dataset:
+    """
+    Initialize the PSET dataset and set the Epoch.
+
+    The Epoch time is set to the mid-point of the L1B
+    Direct Event times. There is one Epoch per PSET file.
+
+    Parameters
+    ----------
+    l1b_de : xarray.Dataset
+        L1B Direct Event dataset.
+    attr_mgr : ImapCdfAttributes
+        Attribute manager used to get the L1C attributes.
+    logical_source : str
+        The logical source of the pset.
+
+    Returns
+    -------
+    pset : xarray.Dataset
+        Initialized PSET dataset.
+    """
+    pset = xr.Dataset(
+        attrs=attr_mgr.get_global_attributes(logical_source),
+    )
+
+    mid_idx = len(l1b_de["epoch"]) // 2
+    pset_epoch = l1b_de["epoch"][mid_idx].item()
+    pset["epoch"] = xr.DataArray(
+        np.array([pset_epoch]),
+        dims=["epoch"],
+        attrs=attr_mgr.get_variable_attributes("epoch"),
+    )
+
+    return pset
+
+
+def filter_goodtimes(l1b_de: xr.Dataset, anc_dependencies: list) -> xr.Dataset:
+    """
+    Filter the L1B Direct Event dataset to only include good times.
+
+    The good times are read from the sweep table ancillary file.
+
+    Parameters
+    ----------
+    l1b_de : xarray.Dataset
+        L1B Direct Event dataset.
+
+    anc_dependencies : list
+        Ancillary files needed for L1C data product creation.
+
+    Returns
+    -------
+    l1b_de : xarray.Dataset
+        Filtered L1B Direct Event dataset.
+    """
+    # Get the sweep table from the ancillary dependencies
+    sweep_table = next(
+        (item for item in anc_dependencies if "sweeptable" in item), None
+    )
+    # sweep table is a dependency so this should always be in the list
+    sweep_table_df = pd.read_csv(sweep_table)
+
+    # convert goodtimes from MET to TTJ2000
+    goodtimes_start = met_to_ttj2000ns(sweep_table_df["GoodTime_strt"])
+    goodtimes_end = met_to_ttj2000ns(sweep_table_df["GoodTime_end"])
+
+    # Create a mask for epochs within any of the start/end time ranges
+    mask = np.zeros_like(l1b_de["epoch"], dtype=bool)
+
+    # Iterate over the good times and create a mask
+    for start, end in zip(goodtimes_start, goodtimes_end):
+        mask |= (l1b_de["epoch"] >= start) & (l1b_de["epoch"] < end)
+
+    # Filter the dataset using the mask
+    filtered_epochs = l1b_de.sel(epoch=mask)
```

Comment:
It looks like this is a "good" mask? If so, could we label this as something like mask_good, mask_selection to indicate it is selecting the good values and not pruning the bad ones? I think that would make it more obvious to me at first glance.